src/rgw/rgw_rest_s3.cc

```diff
@@ -462,6 +467,33 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
       i != attrs.end()) {
     dump_header(s, "x-amz-replication-status", i->second);
   }
+  // restore status
+  if (attr_iter != attrs.end()) {
+    bufferlist bl = attr_iter->second;
+    auto iter = bl.cbegin();
+    decode(restore_status, iter);
+  }
+
+  if (expire_iter != attrs.end()) {
+    bufferlist bl = expire_iter->second;
+    auto iter = bl.cbegin();
+    decode(expiration_date, iter);
+  }
+
+  if (attr_iter == attrs.end() || restore_status == rgw::sal::RGWRestoreStatus::RestoreFailed) {
+    s->err.http_ret = 202; //Accepted
+    dump_header(s, "x-amz-restore", "on-going-request=\"true\"");
+  } else if (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
+    s->err.http_ret = 409; // Conflict
```

Comment:
AWS specs mention about response codes only for Restore Object CLI but not for get-object/read-through. Lets not update http_ret in this routine. 

In addition, we ideally should set `x-amz-restore` only for HEAD request. Can you check if there is any way (might be there) to verify its HEAD request and then set `x-amz-restore` .

Reply:
Okay sure. Will update that then.